diff --git a/input.cpp b/input.cpp
index 1d53ca8..e97c6e1 100644
--- a/input.cpp
+++ b/input.cpp
@@ -1922,6 +1922,23 @@ static void input_lightgun_load(int idx)
 	FileLoadConfig(name, input[idx].guncal, 4 * sizeof(int32_t));
 }
 
+static bool input_is_xgunner(int idx)
+{
+	return input[idx].vid == 0x1209 && input[idx].pid >= 0x0001 && input[idx].pid <= 0x0004;
+}
+
+static void input_xgunner_setup(int idx)
+{
+	input[idx].quirk = QUIRK_LIGHTGUN_MOUSE;
+	input[idx].lightgun = 1;
+	input[idx].num = input[idx].pid & 0xf;
+	input[idx].guncal[0] = 0;
+	input[idx].guncal[1] = 32767;
+	input[idx].guncal[2] = 0;
+	input[idx].guncal[3] = 32767;
+	input_lightgun_load(idx);
+}
+
 int input_has_lightgun()
 {
 	for (int i = 0; i < NUMDEV; i++)
@@ -4218,6 +4235,8 @@ void mergedevs()
 				memcpy(input[i].name, input[j].name, sizeof(input[i].name));
 				memcpy(input[i].idstr, input[j].idstr, sizeof(input[i].idstr));
 
+				if (input_is_xgunner(i)) input_xgunner_setup(i);
+
 				if (!input[i].quirk)
 				{
 					//All mice as spinners
@@ -5437,6 +5456,9 @@ int input_test(int getchar)
 							input_lightgun_load(n);
 						}
 
+						//X-GUNNER LCD lightgun
+						if (input_is_xgunner(n)) input_xgunner_setup(n);
+
 						//Sinden Lightgun (two different PIDs, four different PIDs depending on gun color/config)
 						if ((input[n].vid == 0x16c0 || input[n].vid == 0x16d0) && (
             					input[n].pid == 0x0f01 ||
@@ -5994,7 +6016,7 @@ int input_test(int getchar)
 									}
 								}
 
-								if (ev.type == EV_ABS && (input[i].quirk == QUIRK_LIGHTGUN || input[i].quirk == QUIRK_LIGHTGUN_MOUSE))
+								if (ev.type == EV_ABS && (input[i].quirk == QUIRK_LIGHTGUN || input[i].quirk == QUIRK_LIGHTGUN_MOUSE) && !input_is_xgunner(i))
 								{
 									menu_lightgun_cb(i, ev.type, ev.code, ev.value);
 
diff --git a/menu.cpp b/menu.cpp
index 9577174..e4271f1 100644
--- a/menu.cpp
+++ b/menu.cpp
@@ -1018,6 +1018,7 @@ static int gun_ok = 0;
 static int gun_side = 0;
 static int gun_idx = 0;
 static int32_t gun_pos[4] = {};
+static uint32_t gun_ignore_timer = 0;
 static int page = 0;
 
 static void menu_button_name(int button, char *buf, size_t bsize)
@@ -7501,6 +7502,7 @@ void HandleUI(void)
 		gun_side = 0;
 		gun_y = 0;
 		gun_x = 0;
+		gun_ignore_timer = GetTimer(750);
 		memset(gun_pos, 0, sizeof(gun_pos));
 		menustate = MENU_LGCAL1;
 		menusub = 0;
@@ -7533,6 +7535,7 @@ void HandleUI(void)
 		{
 			gun_ok = 0;
 			gun_side++;
+			gun_ignore_timer = GetTimer(350);
 		}
 
 		if (gun_ok == 2)
@@ -8582,6 +8585,8 @@ int menu_lightgun_cb(int idx, uint16_t type, uint16_t code, int value)
 	{
 		if ((code == 0x130 || code == 0x131 || code == 0x120 || code == 0x110) && menustate == MENU_LGCAL1)
 		{
+			if (gun_ignore_timer && !CheckTimer(gun_ignore_timer)) return 1;
+			gun_ignore_timer = 0;
 			gun_idx = idx;
 			if (value == 1) gun_ok = 1;
 			if (value == 0) gun_ok = 2;
